src/react/hooks/useFetchDaos.ts

```diff
@@ -0,0 +1,36 @@
+import { useContext, useQuery } from './imports';
+import { Dao } from '../../core/types/Dao';
+import { SupportedChainId } from '../../core/types/Chains';
+import { getAllDaos } from '../../core/fetch/dao';
+import { QueryReturn } from '../types';
+import { DecentApiContext } from '../contexts/DecentApiContext';
+
+type UseFetchDaosParams = {
+  chainId?: SupportedChainId;
+};
+
+/**
+ * React hook to fetch all DAOs.
+ *
+ * @param {UseFetchDaosParams} params - Optional parameters to filter DAOs by chainId.
+ * @param {SupportedChainId} params.chainId - The EIP155 chain ID
+ * @returns {QueryReturn<Dao[]>} Object with { data: Dao[], isLoading: boolean, error: Error | null }
+ */
+export const useFetchDaos = (params?: UseFetchDaosParams): QueryReturn<Dao[]> => {
+  const { chainId } = params ?? {};
+  const { apiUrl } = useContext(DecentApiContext);
+
+  const shouldFetch = !!chainId;
```

Comment:
maybe we should support `shouldFetch` or some options of [TanStack Query parameters](https://wagmi.sh/react/api/hooks/useEnsAddress#enabled) within `params` of all hooks. this could be useful for skipping hooks.

Reply:
https://linear.app/decent-labs/issue/ENG-758/add-shouldfetch-bool-to-all-hook-params